service/src/main/java/com/epam/esm/service/service/certificate/CertificateServiceImpl.java

```diff
@@ -224,43 +269,12 @@ private void validateParameterValues(List<String> parameterValues) throws BadReq
         }
     }
 
-    private Certificate findOldCertificate(int certificateId) throws BadRequestException {
-        if (certificateId <= 0) {
-            throw new BadRequestException("Wrong certificate id!", ErrorCodes.CERTIFICATE_BAD_REQUEST);
-        }
-        return dao.getById(certificateId);
-    }
-
-    @Transactional
-    @Override
-    public Certificate patch(Certificate certificate)
-            throws ValidationException, BadRequestException, NotFoundException {
-        Certificate oldCertificate = findOldCertificate(certificate.getId());
-
-        if (oldCertificate == null) {
-            throw new BadRequestException("Wrong certificate id!", ErrorCodes.CERTIFICATE_BAD_REQUEST);
-        }
-        return update(updateCertificate(certificate, oldCertificate));
-    }
-
-    private Certificate updateCertificate(Certificate newCertificate, Certificate oldCertificate) {
-        if (newCertificate.getPrice() != null
-                && newCertificate.getPrice().compareTo(BigDecimal.ZERO) > 0) {
-            oldCertificate.setPrice(newCertificate.getPrice());
-        }
-        if (newCertificate.getDescription() != null) {
-            oldCertificate.setDescription(newCertificate.getDescription());
-        }
-        if (newCertificate.getDuration() != 0) {
-            oldCertificate.setDuration(newCertificate.getDuration());
-        }
-        if (newCertificate.getName() != null) {
-            oldCertificate.setName(newCertificate.getName());
-        }
-        if (CollectionUtils.isNotEmpty(newCertificate.getTags())) {
-            oldCertificate.setTags(newCertificate.getTags());
-        }
-        return oldCertificate;
+    private Certificate findOldCertificate(int certificateId)
+            throws NotFoundException {
+        Optional<Certificate> certificate = dao.findById(certificateId);
+        checkIfPresent(certificate, "id", Integer.toString(certificateId),
```

Comment:
you don't use the returned value here and overall you can use orElseThrow method

Reply:
Agree, it will be longer but clearer